internal/jujuapi/utils.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 Canonical.
+package jujuapi
+
+import (
+	"context"
+	"strings"
+
+	"github.com/juju/names/v5"
+
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+func checkModelAccessForUser(ctx context.Context, path string, u *openfga.User, mt names.ModelTag) (bool, error) {
```

Comment:
The fact this takes a path is confusing and not documented, we're splitting on "/" and taking just [0], it doesn't really explain what you're doing. I also think because it's looking for log/charms/applications it's a mysterious method, log/charms/applications could be anything, it just so happens they're path segments for authorisation. We need to do this better, I'd recommend the middleware approach and register it per each endpoint for the kind of access needed. For example, mux.Use(ModelReaderAuthoriser()) on a handler just for the log endpoint.